account_cash_discount_write_off/models/account_payment_line.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class PaymentLine(models.Model):
+
+    _inherit = 'account.payment.line'
+
+    @api.multi
+    def check_cash_discount_write_off_creation(self):
+        self.ensure_one()
+        move_line = self.move_line_id
+        has_partial_reconciliation = (
+            bool(move_line.matched_debit_ids) or
+            bool(move_line.matched_credit_ids))
+        return (
+            self.pay_with_discount and
+            has_partial_reconciliation and
+            move_line.invoice_id
+        )
```

Comment:
`not move_line.reconciled` instead of `has_partial_reconciliation`?

Reply:
perhaps we should do this writeoff only when the amound paid is exactly the cash discount amount on the invoice?